### PR TITLE
Add support for directories to the Files tab

### PR DESCRIPTION
**Pull request by siggisim**
We currently only show files from the exec log on the Files tab. This change adds support for showing directories too.
Screenshot 2024-12-19 at 1 13 54 PM

### CODE REVIEW

File: app/invocation/invocation_file_card.tsx [hunk 1]
```diff
@@ -111,7 +117,7 @@ export default class InvocationExecLogCardComponent extends React.Component<Prop
     }
   }
 
-  sort(a: tools.protos.ExecLogEntry, b: tools.protos.ExecLogEntry): number {
+  sortFiles(a: tools.protos.ExecLogEntry, b: tools.protos.ExecLogEntry): number {
```
Comment (bduffany) [inline app/invocation/invocation_file_card.tsx:120]: nit: `sortFiles` => `compareFiles`

File: app/invocation/invocation_file_card.tsx [hunk 2]
```diff
@@ -123,6 +129,36 @@ export default class InvocationExecLogCardComponent extends React.Component<Prop
     }
   }
 
+  sortDirectories(a: tools.protos.ExecLogEntry, b: tools.protos.ExecLogEntry): number {
```
Comment (bduffany) [inline app/invocation/invocation_file_card.tsx:132]: nit: `sortDirectories` => `compareDirectories`

File: app/invocation/invocation_file_card.tsx [hunk 3]
```diff
@@ -123,6 +129,36 @@ export default class InvocationExecLogCardComponent extends React.Component<Prop
...
+    return (files || [])
+      .map((f) => f.digest?.hash || "")
+      .reduce((a, b) => {
+        var hash = "";
+        for (var i = 0; i < a.length; i++) {
+          let char = (a.charCodeAt(i) + b.charCodeAt(i)) % 36;
+          hash += String.fromCharCode(char < 26 ? char + 97 : char + 22);
+        }
+        return hash;
+      });
```
Comment (bduffany) [inline app/invocation/invocation_file_card.tsx:159]: need an initial value for the case where the list is empty:

```suggestion
      }, "");
```

**Review by bduffany — APPROVED** [review]

File: app/invocation/invocation_file_card.tsx (hunk 1 above)
Comment (siggisim) [reply app/invocation/invocation_file_card.tsx]: Done

File: app/invocation/invocation_file_card.tsx (hunk 2 above)
Comment (siggisim) [reply app/invocation/invocation_file_card.tsx]: Done

File: app/invocation/invocation_file_card.tsx (hunk 3 above)
Comment (siggisim) [reply app/invocation/invocation_file_card.tsx]: Done

**Pull request merged**